xwiki-platform-core/xwiki-platform-store/xwiki-platform-store-filesystem-oldcore/src/main/java/org/xwiki/store/filesystem/internal/DefaultTemporaryAttachmentSessionsManager.java

```diff
@@ -58,21 +61,18 @@ public class DefaultTemporaryAttachmentSessionsManager implements TemporaryAttac
     @Inject
     private Provider<XWikiContext> contextProvider;
 
+    @Inject
+    private AttachmentValidator attachmentValidator;
```

Comment:
Does not need to be in a `Provider` AFAIR the `DefaultTemporaryAttachmentSessionManager` is injected quite early no?

Reply:
This is an annotation provided by jsr311, this is part of its extensibility mechanism. It's used to scan annotations to discover "components".

Reply:
I did not notice any issue with that yet.